File: quarkus/manager-service/src/main/java/com/javanc/manager/domain/model/Job.java
package com.javanc.manager.domain.model;

import io.quarkus.mongodb.panache.common.MongoEntity;
import org.bson.codecs.pojo.annotations.BsonId;

import java.util.List;

@MongoEntity(collection = "job")
public class Job {
    @BsonId
    public Integer id;
    public String title;
    public String description;
    public TypeJob typeJob;
    public Integer size;
    public List<Integer> idProfiePending;
    public List<Integer> idProfile;
    public Integer idCompany;
}

File: client-react/src/types/job.ts
export type TypeJob = "java" | "python" | "php";

export interface JobDTO {
  id: number;
  title: string;
  description?: string;
  typeJob?: TypeJob;
  size?: number;
  idProfiePending?: number[];
  idProfile?: number[];
  idCompany: number;
}

export type JobApplicationState =
  | "accepted"
  | "pending"
  | "closed"
  | "open";

export interface JobFormValues {
  title: string;
  description?: string;
  typeJob: TypeJob;
  size: number;
}
